### DISCUSSION TITLE
Re-naming OAR labels on a segmentation model after training

### DISCUSSION THREAD

**Discussion by alexconstant** [microsoft/InnerEye-DeepLearning#764] [Q&A]
Hi All,

I was just wondering if there was a way to re-name the labels of the OARs in a segmentation model after training is complete?

Thank you!

Alex

**Comment by ant0nsc** [comment]
Hi @alexconstant , yes, I think that is possible but it is a bit fiddly.

Structure names (and colors) are added only during inference, when we convert from a segmentation mask to DicomRT. 

The colors are defined in the model field `colours`, structure names are defined in `ground_truth_ids_display_names`. Check where that is defined, it may be in your derived class, or you may need to go back to the base class. Once you know where to modify, you would need to
- download the trained model from AzureML, probably best via the script in [InnerEye/Scripts/move_model](https://github.com/microsoft/InnerEye-DeepLearning/blob/main/InnerEye/Scripts/move_model.py)
- In the downloaded model, there is a full copy of the InnerEye codebase. Modify the code in-place and put in the new names and/or colours
- Re-upload the model to AzureML. This will create a new model version.

Let me know if that works!

**Reply by alexconstant** [reply]
Thanks @ant0nsc! 

I'm assuming that changing the names of the OARs can be done in the same way as changing the colours?

Thanks

**Reply by ant0nsc** [reply]
yes, absolutely, I added to the answer.

**Reply by alexconstant** [reply]
Thank you!
